tensorflow_model_optimization/g3doc/guide/clustering/clustering_comprehensive_guide.ipynb

```diff
@@ -0,0 +1,433 @@
+{
```

Comment:
<p>Generally take a look at the quantization aware training and updated pruning for style, etc. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/tensorflow/model-optimization/pull/330/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

Reply:
<p>OK, can we agree on the outline of the comprehensive guide, since I deleted several sections? </p><ol><li>prune custom Keras layer or modify parts of layer to cluster (not sure whether it would apply to clustering)</li><li>train model, model.fit, custom training loop (deleted them since we do not have the callbacks in clustering afaik)</li><li>more readable but potentially lower model accuracy (deleted because clustering only works on pretrained model and I had problems of intializing the weights in the process of defining the models)</li><li>hardware-specific optimizations (same as above, not sure whether it applies to clustering)</li></ol><p> </p><p>Also, is it necessary to have tensorboard?</p>

Reply:
1. - 4. are all not applicable to clustering.

1. because you don't have the equivalent of ClusterableLayer in the first release yet
2. and 3. for the reasons as you said
4. because you don't have any APIs for clustering yet that I know of.

I don't see the value of TensorBoard here since the e2e example is for demonstrating accuracy results, not this comprehensive guide. You also don't have the equivalent of `PruningSummaries` either, where Tensorboard is useful to demonstrate.

Reply:
<blockquote>'&nbsp;you don't have the equivalent of ClusterableLayer in the first release yet' </blockquote><p>??</p><p> </p><blockquote>I don't see the value of TensorBoard here since the e2e example is for demonstrating accuracy results, not this comprehensive guide</blockquote><p>Will add it for the end to end example.</p><p> </p><p> </p>

Reply:
> you don't have the equivalent of ClusterableLayer in the first release yet'

Sorry. Meant the equivalent of PrunableLayer.

Reply:
Also - since you don't have the equivalent of `PruningSummaries`, actually there isn't a huge benefit to adding it to the e2e example either. The training/eval progress bar is already shown.